emperor/pandas.py

```diff
@@ -0,0 +1,116 @@
+the Emperor.
+
+.. currentmodule:: emperor.pandas
+
+Functions
+---------
+.. autosummary::
+    :toctree: generated/
+
+    scatterplot
+"""
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, emperor development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE.md, distributed with this software.
+# ----------------------------------------------------------------------------
+from __future__ import division
+
+import numpy as np
+import pandas as pd
+
+from emperor.core import Emperor
+from skbio import OrdinationResults
+
+
+def scatterplot(df, x=None, y=None, z=None, remote=True):
+    """Create an Emperor scatter plot from a Pandas DataFrame
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        Pandas DataFrame with the data to display
+    x, y, z : str
+        Column names in `df`, these will represent the order of the axes in the
+        final visualization.
+    remote : bool
+        Whether the JavaScript resources should be loaded locally or from
+        GitHub
+
+    Returns
+    -------
+    emperor.core.Emperor
+        Emperor object with the numerical data as the `ordination` attribute
+        and the entire DataFrame as the `mf` attribute.
+
+    Raises
+    ------
+    ValueError
+        If `df` is not a PandasDataFrame
+        If `x`, `y` or `z` are missing from `df` or if they are not numeric
+        columns.
+        If after removing rows with missing data there are fewer than 3
+        samples.
+
+    Notes
+    -----
+    If a row has missing data, that data pont will be removed from the
+    visualization.
+
+    See Also
+    --------
+    emperor.core.Emperor
+    """
+
+    if not isinstance(df, pd.DataFrame):
+        raise ValueError("The argument is not a Pandas DataFrame")
+
+    for col in [z, y, x]:
+        if col is None:
+            continue
+
+        if col not in df.columns:
+            raise ValueError("'%s' is not a column in the DataFrame" % col)
+
+        if not np.issubdtype(df[col].dtype, np.number):
+            raise ValueError("'%s' is not a numeric column" % col)
+
+    # remove NAs
+    samples = df.select_dtypes(include=[np.number]).copy()
+    samples.dropna(axis=0, how='any', inplace=True)
+
+    if len(samples.columns) < 3:
+        raise ValueError("Not enough data to plot")
+
+    # sort columns by variance
+    variance = samples.var().sort_values(ascending=False)
+    samples = samples[variance.index].copy()
+
+    # re-order x, y and z
+    ordered = samples.columns.tolist()
+    for col in [z, y, x]:
+        if col is not None:
+            ordered.remove(col)
+            ordered = [col] + ordered
+    samples = samples[ordered].copy()
+
+    # match up the metadata and coordinates
+    df = df.loc[samples.index].copy()
+
+    ores = OrdinationResults(short_method_name='Ax', long_method_name='Axis',
```

Comment:
Sounds like a good idea, though I'm really not sure if we should even be
exposing/passing these arguments, mainly because this is not intended to
be used as an "ordination" interface ... but now, I don't know, perhaps
the best would be to just not pass anything here(?).

On (Jul-10-16|23:00), Jamie Morton wrote:
> > +    # sort columns by variance
> > +    variance = samples.var().sort_values(ascending=False)
> > +    samples = samples[variance.index].copy()
> > +
> > +    # re-order x, y and z
> > +    ordered = samples.columns.tolist()
> > +    for col in [z, y, x]:
> > +        if col is not None:
> > +            ordered.remove(col)
> > +            ordered = [col] + ordered
> > +    samples = samples[ordered].copy()
> > +
> > +    # match up the metadata and coordinates
> > +    df = df.loc[samples.index].copy()
> > +
> > +    ores = OrdinationResults(short_method_name='Ax', long_method_name='Axis',
> 
> What do you think about passing in `short_method_name` and `long_method_name` as optional parameters?
> 
> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/emperor/pull/517/files/911c1dc15f3025f6d79dc65dee8bec576c7599d9#r70205320